src/NServiceBus.AcceptanceTests/Serialization/When_sanitizing_xml_messages.cs

```diff
@@ -0,0 +1,85 @@
+﻿namespace NServiceBus.AcceptanceTests.Serialization
+{
+    using System.Text;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using MessageMutator;
+    using NUnit.Framework;
+
+    public class When_sanitizing_xml_messages : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_remove_illegal_characters_from_messages()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointSanitizingInput>(e => e
+                    .When(session => session.SendLocal(new SimpleMessage {Value = "Hello World!"})))
+                .Done(c => c.MessageReceived)
+                .Run();
+
+            Assert.That(context.Input, Is.EqualTo("Hello World!"));
+        }
+
+        class Context : ScenarioContext
+        {
+            public bool MessageReceived { get; set; }
+            public string Input { get; set; }
+        }
+
+        class EndpointSanitizingInput : EndpointConfigurationBuilder
+        {
+            public EndpointSanitizingInput()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.UseSerialization<XmlSerializer>().SanitizeInput();
+                    c.RegisterComponents(r => r.ConfigureComponent<IncomingMutator>(DependencyLifecycle.SingleInstance));
+                });
+            }
+
+            class SimpleMessageHandler : IHandleMessages<SimpleMessage>
+            {
+                public SimpleMessageHandler(Context scenarioContext)
+                {
+                    this.scenarioContext = scenarioContext;
+                }
+
+                public Task Handle(SimpleMessage message, IMessageHandlerContext context)
+                {
+                    scenarioContext.MessageReceived = true;
+                    scenarioContext.Input = message.Value;
+
+                    return Task.FromResult(0);
+                }
+
+                Context scenarioContext;
+            }
+
+            public class IncomingMutator : IMutateIncomingTransportMessages
+            {
+                public IncomingMutator(Context scenarioContext)
+                {
+                    this.scenarioContext = scenarioContext;
+                }
+
+                public Task MutateIncoming(MutateIncomingTransportMessageContext context)
+                {
+                    var body = Encoding.UTF8.GetString(context.Body);
+                    var invalidChar = char.ConvertFromUtf32(0x8);
```

Comment:
I have no idea what character that is tbh. I just took one which was on the filtered chars. Apparently it's not very easy to sneak in one of the chars above 0x0Fxxxx(whatever that range was) since these can't be converted to chars (while 0x8 can). Otherwise we would need to replace raw bytes.